p2p/src/connection/outgoing/p2p_connection_outgoing_reducer.rs

```diff
@@ -488,40 +542,50 @@ impl P2pConnectionOutgoingState {
                     .outgoing_peer_connection_mut(&peer_id)
                     .ok_or_else(|| format!("Invalid state: {:?}", action))?;
 
-                if let Self::FinalizeSuccess {
+                let Self::FinalizeSuccess {
                     offer,
                     answer,
                     rpc_id,
+                    on_success,
                     ..
                 } = state
-                {
-                    *state = Self::Success {
-                        time,
-                        offer: offer.clone(),
-                        answer: answer.clone(),
-                        rpc_id: rpc_id.take(),
-                    };
-                } else {
+                else {
                     bug_condition!(
                         "Invalid state for `P2pConnectionOutgoingAction::Success`: {:?}",
                         state
                     );
                     return Ok(());
-                }
+                };
+
+                // remove callback from state if it exists and call it, this is so same callback doesn't get called repeatedly
```

Comment:
@0xMimir this is more about retaining the callback so that it can be used later right? the previous state is discarded on the next line anyway. Lets get rid of the comment because it adds confusion